src/Microsoft.DotNet.ProjectModel/Files/IncludeContext.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.ProjectModel.Files
+{
+    public class IncludeContext
+    {
+        private static readonly char[] PatternSeparator = new[] { ';' };
+
+        public IncludeContext(
+            string sourceBasePath,
+            string option,
+            JObject rawObject,
+            string[] defaultBuiltInInclude = null,
+            string[] defaultBuiltInExclude = null)
```

Comment:
let's avoid the default parameter. this is an internal type, let make the client send in `null`